lib/core/Canvas.js

```diff
@@ -369,38 +369,42 @@ Canvas.prototype.getPlane = function(name) {
  * Creates a plane that is used to draw elements on it. If no
  * root element is provided, an implicit root will be used.
  *
- * @param {string} name
+ * @param {string|Object} plane
  * @param {Object|djs.model.Root} [rootElement] optional root element
  *
  * @return {Object} plane descriptor with { layer, rootElement, name }
  */
-Canvas.prototype.createPlane = function(name, rootElement) {
-  if (!name) {
-    throw new Error('must specify a name');
+Canvas.prototype.createPlane = function(plane, rootElement) {
+  if (!plane) {
+    throw new Error('must specify a plane');
+  }
+
+  if (this._planes[plane] || this._planes[plane.name]) {
+    throw new Error('plane <' + plane + '> already exists');
+  }
+
+  if (typeof plane === 'string') {
+    plane = { name: plane };
   }
 
-  if (this._planes[name]) {
-    throw new Error('plane <' + name + '> already exists');
+  if (!plane.layer) {
+    var svgLayer = this.getLayer(plane.name, PLANE_LAYER_INDEX);
+    svgAttr(svgLayer, 'display', 'none');
+
+    plane.layer = svgLayer;
   }
 
-  if (!rootElement) {
-    rootElement = {
-      id: '__implicitroot' + name,
+  if (!plane.rootElement) {
+    plane.rootElement = rootElement || {
+      id: '__implicitroot' + plane.name,
       children: [],
       isImplicit: true
     };
   }
 
-  var svgLayer = this.getLayer(name, PLANE_LAYER_INDEX);
-  svgAttr(svgLayer, 'display', 'none');
+  this._addRoot(plane.rootElement, plane);
```

Comment:
I like this very much, as if we're already transitioning to https://github.com/bpmn-io/diagram-js/issues/600. :wink: :heart: